src/org/zaproxy/zap/extension/alert/PopupMenuItemAlert.java

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.alert;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.Component;
+import java.util.HashSet;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.TreePath;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.control.Control;
+import org.parosproxy.paros.extension.ExtensionPopupMenuItem;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.zaproxy.zap.extension.alert.ExtensionAlert;
+
+
+public abstract class PopupMenuItemAlert extends ExtensionPopupMenuItem {
+
+	private static final long serialVersionUID = 1L;
+    private final boolean multiSelect;
+    private final ExtensionAlert extAlert;
+
+    private static final Logger log = Logger.getLogger(PopupMenuItemAlert.class);
+
+    public PopupMenuItemAlert(String label) {
+        this(label, false);
+    }
+
+	public PopupMenuItemAlert(String label, boolean multiSelect) {
+		super(label);
+        this.multiSelect = multiSelect;
+        addActionListener(new PerformActionsActionListener());
+        this.extAlert = Control.getSingleton().getExtensionLoader().getExtension(ExtensionAlert.class);
+	}
+
+    public final boolean isMultiSelect() {
+        return multiSelect;
+    }
+
+    private HashSet<Alert> getAlertNodes() {
+    	TreePath[] paths = this.extAlert.getAlertPanel().getTreeAlert().getSelectionPaths();
+        HashSet<Alert> alertNodes = new HashSet<Alert>();
+        for(int i = 0; i < paths.length; i++ ) {
+            DefaultMutableTreeNode alertNode = (DefaultMutableTreeNode)paths[i].getLastPathComponent();
+            System.out.println((Alert)alertNode.getUserObject());
+            if(alertNode.getChildCount() == 0) {
+                alertNodes.add(((Alert)alertNode.getUserObject()));
+                continue;
+            }
+            for(int j = 0; j < alertNode.getChildCount(); j++ ) {
+                DefaultMutableTreeNode node = (DefaultMutableTreeNode)alertNode.getChildAt(j);
+                alertNodes.add(((Alert)node.getUserObject()));
+            }
+        }
+        return alertNodes;
+    }
+
+    protected abstract void performAction(Alert alert);
+
+    protected void performActions(HashSet<Alert> alerts) {
+        for(Alert alert: alerts) {
+            performAction(alert);
+        }
+    }
+
+    protected boolean isButtonEnabledForNumberOfSelectedAlerts(int count) {
+        if(count == 0 ) {
+            return false;
+        } else if(!isMultiSelect() && count>1 ) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public boolean isEnableForComponent(Component invoker) {
+        TreePath[] paths = this.extAlert.getAlertPanel().getTreeAlert().getSelectionPaths();
+        for(int i = 0; i < paths.length; i++ ){
```

Comment:
The only node that does not have a user object is the root node so it can be directly checked, for example:
```Java
this.extAlert.getAlertPanel().getTreeAlert().isRowSelected(0)
```

This check should be done for the enabled state of the menu item (below when calling setEnabled) not to decide whether the menu is shown or not (they should still be shown in the alerts tree but be disabled).